hidimstat/clustered_inference.py

```diff
@@ -1,91 +1,204 @@
 
-    sf_compressed = ward.inverse_transform(sf)
-    sf_corr_compressed = ward.inverse_transform(sf_corr)
-    cdf_compressed = ward.inverse_transform(cdf)
-    cdf_corr_compressed = ward.inverse_transform(cdf_corr)
+        raise ValueError('Unknow method')
 
-    if predict:
+    return beta_hat, sf, sf_corr, cdf, cdf_corr
 
-        labels = ward.labels_
-        clusters_size = np.zeros(labels.size)
 
-        for label in range(labels.max() + 1):
-            cluster_size = np.sum(labels == label)
-            clusters_size[labels == label] = cluster_size
+def _degrouping(ward, beta_hat, sf, sf_corr, cdf, cdf_corr):
+    """Assigning cluster-wise stats to features contained in the corresponding
+    cluster and rescaling estimated parameter"""
 
-        beta_hat_compressed = \
-            ward.inverse_transform(beta_hat) / clusters_size
+    sf_degrouped = ward.inverse_transform(sf)
+    sf_corr_degrouped = ward.inverse_transform(sf_corr)
+    cdf_degrouped = ward.inverse_transform(cdf)
+    cdf_corr_degrouped = ward.inverse_transform(cdf_corr)
 
-        return (sf_compressed, sf_corr_compressed, cdf_compressed,
-                cdf_corr_compressed, beta_hat_compressed)
+    labels = ward.labels_
+    clusters_size = np.zeros(labels.size)
 
-    return (sf_compressed, sf_corr_compressed, cdf_compressed,
-            cdf_corr_compressed)
+    for label in range(labels.max() + 1):
+        cluster_size = np.sum(labels == label)
+        clusters_size[labels == label] = cluster_size
 
+    if len(beta_hat.shape) == 1:
 
-def hd_inference(X, y, method='DL', predict=False, n_jobs=1):
+        beta_hat_degrouped = ward.inverse_transform(beta_hat) / clusters_size
 
-    if method == 'DL':
+    elif len(beta_hat.shape) == 2:
 
-        beta_hat, cb_min, cb_max = \
-            desparsified_lasso_confint(X, y, n_jobs=n_jobs)
-        sf, sf_corr = sf_from_cb(cb_min, cb_max)
-        cdf, cdf_corr = cdf_from_cb(cb_min, cb_max)
+        n_features = sf_degrouped.shape[0]
+        n_times = beta_hat.shape[1]
+        beta_hat_degrouped = np.zeros((n_features, n_times))
 
-    else:
-        raise ValueError('Unknow method')
+        for i in range(n_times):
+
+            beta_hat_degrouped[:, i] = \
+                ward.inverse_transform(beta_hat[:, i]) / clusters_size
+
+    return (beta_hat_degrouped, sf_degrouped, sf_corr_degrouped,
+            cdf_degrouped, cdf_corr_degrouped)
+
+
+def clustered_inference(X_init, y, ward, n_clusters, train_size=0.3,
+                        groups=None, method='desparsified-lasso', seed=0,
+                        n_jobs=1, memory=None, verbose=1, **kwargs):
+    """Clustered inference algorithm
+
+    Parameters
+    ----------
+    X_init : ndarray, shape (n_samples, n_features)
+        Original data (uncompressed).
+
+    y : ndarray, shape (n_samples,) or (n_samples, n_times)
+        Target.
+
+    ward : sklearn.cluster.FeatureAgglomeration Object
```

Comment:
```suggestion
    ward : sklearn.cluster.FeatureAgglomeration
```

Reply:
you don't want to have a default for this parameter?

Reply:
No because, you need to know the connectivity matrix and there is no default for that... WDYT ?